archive/2023-03-11-DssSpell/DssSpell.sol

```diff
@@ -0,0 +1,199 @@
+    function delay() external view returns (uint256);
+    function plot(address, bytes32, bytes calldata, uint256) external;
+    function exec(address, bytes32, bytes calldata, uint256) external returns (bytes memory);
+    function setDelay(uint256) external;
+}
+
+interface Changelog {
+    function getAddress(bytes32) external view returns (address);
+}
+
+interface SpellAction {
+    function officeHours() external view returns (bool);
+    function description() external view returns (string memory);
+    function nextCastTime(uint256) external view returns (uint256);
+}
+
+interface DirectDepositMomAbstract {
+    function disable(address) external;
+}
+
+contract DssExec {
+
+    Changelog      constant public log   = Changelog(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+    uint256                 public eta;
+    bytes                   public sig;
+    bool                    public done;
+    bytes32       immutable public tag;
+    address       immutable public action;
+    uint256       immutable public expiration;
+    PauseAbstract immutable public pause;
+
+    // Provides a descriptive tag for bot consumption
+    // This should be modified weekly to provide a summary of the actions
+    // Hash: seth keccak -- "$(wget https://<executive-vote-canonical-post> -q -O - 2>/dev/null)"
+    function description() external view returns (string memory) {
+        return SpellAction(action).description();
+    }
+
+    function officeHours() external view returns (bool) {
+        return SpellAction(action).officeHours();
+    }
+
+    function nextCastTime() external view returns (uint256 castTime) {
+        return SpellAction(action).nextCastTime(eta);
+    }
+
+    // @param _description  A string description of the spell
+    // @param _expiration   The timestamp this spell will expire. (Ex. block.timestamp + 30 days)
+    // @param _spellAction  The address of the spell action
+    constructor(uint256 _expiration, address _spellAction) {
+        pause       = PauseAbstract(log.getAddress("MCD_PAUSE"));
+        expiration  = _expiration;
+        action      = _spellAction;
+
+        sig = abi.encodeWithSignature("execute()");
+        bytes32 _tag;                    // Required for assembly access
+        address _action = _spellAction;  // Required for assembly access
+        assembly { _tag := extcodehash(_action) }
+        tag = _tag;
+    }
+
+    function schedule() public {
+        require(block.timestamp <= expiration, "This contract has expired");
+        require(eta == 0, "This spell has already been scheduled");
+        eta = block.timestamp + PauseAbstract(pause).delay();
+        pause.plot(action, tag, sig, eta);
+
+        DirectDepositMomAbstract(log.getAddress("DIRECT_MOM")).disable(
+            log.getAddress("DIRECT_COMPV2_DAI_PLAN")
+        );
+
+        DirectDepositMomAbstract(log.getAddress("DIRECT_MOM")).disable(
+            log.getAddress("DIRECT_AAVEV2_DAI_PLAN")
+        );
```

Comment:
Is there an intention to unwind these by hand? IIRC `disable` only sets `bar` to zero, still need a call to `exec` to withdraw the liquidity.